arch/arm/soc/nordic_nrf/nrf52/Kconfig.soc

```diff
@@ -223,5 +228,5 @@ config GPIO_AS_PINRESET
 
 config NRF_ENABLE_ICACHE
 	bool "Enable the instruction cache (I-Cache)"
-	depends on SOC_SERIES_NRF52X
+	depends on SOC_NRF52832 || SOC_NRF52840
```

Comment:
The same should be done for `config NFCT_PINS_AS_GPIOS` - this option does not make sense for nRF52810.

Reply:
Will do.